tests/integration/image_checker.py

```diff
@@ -0,0 +1,345 @@
+        os.mkdir(diff_dir)
+
+    bbox = diff.getbbox()
+    if not bbox:
+        # If `diff.getbbox()` is None, then the images are in theory equal
+        assert diff.getbbox() is None
+    else:
+        # Sometimes, a few pixels will differ, but the two images appear identical.
+        # https://codereview.stackexchange.com/questions/55902/fastest-way-to-count-non-zero-pixels-using-python-and-pillow
+        nonzero_pixels = (
+            diff.crop(bbox)
+            .point(lambda x: 255 if x else 0)
+            .convert("L")
+            .point(bool)
+            .getdata()
+        )
+        num_nonzero_pixels = sum(nonzero_pixels)
+        width, height = expected_png.size
+        num_pixels = width * height
+        fraction = num_nonzero_pixels / num_pixels
+        # Fraction of mismatched pixels should be less than 0.02%
+        if fraction >= 0.0002:
+            verbose = False
+            if verbose:
+                print("\npath_to_actual_png={}".format(path_to_actual_png))
+                print("path_to_expected_png={}".format(path_to_expected_png))
+                print("diff has {} nonzero pixels.".format(num_nonzero_pixels))
+                print("total number of pixels={}".format(num_pixels))
+                print("num_nonzero_pixels/num_pixels fraction={}".format(fraction))
+
+            mismatched_images.append(image_name)
+
+            diff_dir_actual_png = os.path.join(
+                diff_dir, "{}_actual.png".format(image_name)
+            )
+            # image_name could contain a number of subdirectories.
+            os.makedirs(os.path.dirname(diff_dir_actual_png), exist_ok=True)
+            shutil.copy(
+                path_to_actual_png,
+                diff_dir_actual_png,
+            )
+            diff_dir_expected_png = os.path.join(
+                diff_dir, "{}_expected.png".format(image_name)
+            )
+            # image_name could contain a number of subdirectories.
+            os.makedirs(os.path.dirname(diff_dir_expected_png), exist_ok=True)
+            shutil.copy(
+                path_to_expected_png,
+                diff_dir_expected_png,
+            )
+            # Draw red box around diff-area on each of: diff, actual, expected
+            _draw_box(diff, diff, os.path.join(diff_dir, f"{image_name}_diff.png"))
+            _draw_box(
+                actual_png, diff, os.path.join(diff_dir, f"{image_name}_actual.png")
+            )
+            _draw_box(
+                expected_png, diff, os.path.join(diff_dir, f"{image_name}_expected.png")
+            )
+
+
+def _draw_box(image, diff, output_path: str):
+    # https://stackoverflow.com/questions/41405632/draw-a-rectangle-and-a-text-in-it-using-pil
+    draw = ImageDraw.Draw(image)
+    (left, upper, right, lower) = (
+        diff.getbbox()
+    )  # We specifically want the diff's bounding box
+    draw.rectangle(((left, upper), (right, lower)), outline="red")
+    image.save(output_path, "PNG")
```

Comment:
This is a little new (and more properly part of the #720 PR). Previously, the red bounding box was only drawn on the `_diff`, but here we draw it on the `_actual` and `_expected` images as well. 

(I haven't actually gotten a mismatched image while running this version of the code, so it's not fully tested. I might try to force a comparison to an obviously different image to check the behavior.)

Reply:
Test has been added as a third commit to #720: 5cf09af02975b26a31bab658371ec04298ec4c16. Bounding box appears on actual, expected, and diff (note this means the actual & expected in the image diff dir, not the original files).